src/app/portlets/dot-edit-page/layout/dot-edit-layout-designer/dot-edit-layout-designer.component.ts

```diff
@@ -0,0 +1,200 @@
+     */
+    isLayout(): boolean {
+        return this.pageView.template.anonymous;
+    }
+
+    /**
+     * Handle the change when user update save as template checkbox value
+     *
+     * @memberof DotEditLayoutDesignerComponent
+     */
+    saveAsTemplateHandleChange(value: boolean): void {
+        const titleFormControl = this.form.get('title');
+        titleFormControl.markAsUntouched();
+        titleFormControl.markAsPristine();
+
+        this.saveAsTemplate = value;
+
+        if (this.saveAsTemplate) {
+            titleFormControl.setValidators(Validators.required);
+            /*
+                Need the timeout so the textfield it's loaded in the DOM before focus, wasn't able to find a better solution
+            */
+            setTimeout(() => {
+                this.templateName.nativeElement.focus();
+            }, 0);
+        } else {
+            titleFormControl.setValidators(null);
+            if (this.initialFormValue.title === null) {
+                titleFormControl.setValue(null);
+            }
+        }
+        titleFormControl.updateValueAndValidity();
+    }
+
+    /**
+     * Get the LayoutBody and call the service to save the layout
+     *
+     * @memberof DotEditLayoutDesignerComponent
+     */
+    saveLayout(event): void {
+        this.dotGlobalMessageService.loading(this.dotMessageService.get('dot.common.message.saving'));
+        const dotLayout: DotLayout = this.form.value;
+        this.pageViewService.save(this.pageView.page.identifier, dotLayout).subscribe(
+            response => {
+                // TODO: This extra request will change once the this.pageViewService.save return a DotPageView object.
+                this.pageViewService.get(this.route.snapshot.queryParams.url).subscribe((pageView: DotPageView) => {
+                    this.dotGlobalMessageService.display(this.dotMessageService.get('dot.common.message.saved'));
+                    this.setupLayout(pageView);
+                });
+            },
+            (err: ResponseView) => {
+                this.dotGlobalMessageService.error(err.response.statusText);
+            }
+        );
+    }
+
+    /**
+     * Set component to edit layout mode, template have no name.
+     *
+     * @memberof DotEditLayoutDesignerComponent
+     */
+    setEditLayoutMode(): void {
+        this.initialFormValue.title = null;
+        this.form.get('title').setValue(null);
+        this.showTemplateLayoutSelectionDialog = false;
+        this.pageView.template.anonymous = true;
+    }
+
+    private setupLayout(pageView?: DotPageView): void {
+        if (pageView) {
+            this.pageView = pageView;
+        }
+        this.templateContainersCacheService.set(this.pageView.containers);
+        this.initForm();
+        this.saveAsTemplateHandleChange(false);
+
+        // Emit event to redraw the grid when the sidebar change
+        this.form.get('layout.sidebar').valueChanges.subscribe(() => {
```

Comment:
I move this manually because it was giving me conflicts with latest @hmoreras changes.